lib/30XUtils/inputValidation.js

```diff
@@ -9,35 +10,36 @@ module.exports = {
   validateSpec: function (spec) {
 
     // Checking for the all the required properties in the specification
-    if (!spec.hasOwnProperty('openapi')) {
+    if (_.isNil(spec.openapi)) {
       return {
         result: false,
         reason: 'Specification must contain a semantic version number of the OAS specification'
       };
     }
 
-    if (!spec.hasOwnProperty('paths')) {
+    if (_.isNil(spec.paths)) {
       return {
         result: false,
         reason: 'Specification must contain Paths Object for the available operational paths'
       };
     }
 
-    if (!spec.hasOwnProperty('info')) {
+    if (_.isNil(spec.info)) {
       return {
         result: false,
         reason: 'Specification must contain an Info Object for the meta-data of the API'
       };
     }
+
     if (!spec.info.hasOwnProperty('$ref')) {
-      if (!spec.info.hasOwnProperty('title')) {
+      if (_.isNil(spec.info.title)) {
```

Comment:
@LuisTejedaS instead of accessing `spec.info.title` directly, we can use `_.get(spec, 'info.title')`. As accessing `spec.info.title` will throw Type error in case of if `spec.info` is Integer or boolean etc.

Same can be case for below `spec.info.version` aceess.

Reply:
Also for other versions, we can also update it. I don't see swagger 2.0 related change here, can we also make sure that it's also addressed.

Reply:
Added use of lodash and validations for swagger 2.0